src/esp/gfx/PbrDrawable.cpp

```diff
@@ -25,48 +28,55 @@ PbrDrawable::PbrDrawable(scene::SceneNode& node,
-          shaderManager.get<MaterialData, PbrMaterialData>(materialDataKey)},
+          shaderManager.get<Magnum::Trade::MaterialData>(materialDataKey)},
       pbrIbl_(pbrIbl) {
-  if (materialData_->metallicTexture && materialData_->roughnessTexture) {
+  if (materialData_->hasAttribute("metallicTexturePtr") &&
+      materialData_->hasAttribute("roughnessTexturePtr")) {
     CORRADE_ASSERT(
-        materialData_->metallicTexture == materialData_->roughnessTexture,
+        materialData_->attribute<Mn::GL::Texture2D*>("metallicTexturePtr") ==
+            materialData_->attribute<Mn::GL::Texture2D*>("roughnessTexturePtr"),
         "PbrDrawable::PbrDrawable(): if both the metallic and roughness "
         "texture exist, they must be packed in the same texture based on glTF "
         "2.0 Spec.", );
   }
+  const auto& tmpMaterialData =
+      materialData_->as<Mn::Trade::PbrMetallicRoughnessMaterialData>();
 
   flags_ = PbrShader::Flag::ObjectId;
-  if (materialData_->textureMatrix != Mn::Matrix3{}) {
+  if (tmpMaterialData.commonTextureMatrix() != Mn::Matrix3{}) {
     flags_ |= PbrShader::Flag::TextureTransformation;
   }
-  if (materialData_->baseColorTexture) {
+  if (materialData_->hasAttribute("baseColorTexturePtr")) {
     flags_ |= PbrShader::Flag::BaseColorTexture;
   }
-  if (materialData_->roughnessTexture) {
+  if (materialData_->hasAttribute("roughnessTexturePtr")) {
     flags_ |= PbrShader::Flag::RoughnessTexture;
   }
-  if (materialData_->metallicTexture) {
+  if (materialData_->hasAttribute("metallicTexturePtr")) {
     flags_ |= PbrShader::Flag::MetallicTexture;
   }
-  if (materialData_->normalTexture) {
+  if (materialData_->hasAttribute("normalTexturePtr")) {
     flags_ |= PbrShader::Flag::NormalTexture;
     if (meshAttributeFlags & gfx::Drawable::Flag::HasTangent) {
       flags_ |= PbrShader::Flag::PrecomputedTangent;
     }
-    if (materialData_->normalTextureScale != 1.0f) {
+    if (tmpMaterialData.attribute<float>(
+            Mn::Trade::MaterialAttribute::NormalTextureScale) != 1.0f) {
       flags_ |= PbrShader::Flag::NormalTextureScale;
-      CORRADE_ASSERT(materialData_->normalTextureScale > 0.0f,
-                     "PbrDrawable::PbrDrawable(): the normal texture scale "
-                     "must be positive.", );
+      CORRADE_ASSERT(
+          tmpMaterialData.attribute<float>(
+              Mn::Trade::MaterialAttribute::NormalTextureScale) > 0.0f,
+          "PbrDrawable::PbrDrawable(): the normal texture scale "
+          "must be positive.", );
     }
   }
-  if (materialData_->emissiveTexture) {
+  if (materialData_->hasAttribute("emissiveTexturePtr")) {
     flags_ |= PbrShader::Flag::EmissiveTexture;
   }
-  if (materialData_->perVertexObjectId) {
+  if (materialData_->attribute<bool>("hasPerVertexObjectId")) {
     // TODO: may be supported in the future
   }
-  if (materialData_->doubleSided) {
+  if (materialData_->isDoubleSided()) {
     flags_ |= PbrShader::Flag::DoubleSided;
```

Comment:
Hm, you could maybe wire this flag in for the Phong drawable now as well. Not sure if you have [that Magnum update](https://github.com/mosra/magnum/commit/c1239b66199926c610cd863e18a4d27a4cef67f1) here tho.

Reply:
@mosra Yeah, it appears we don't have the update supporting this yet.